drivers/serial/uart_xlnx_ps.c

```diff
@@ -313,6 +387,457 @@ static void uart_xlnx_ps_poll_out(struct device *dev, unsigned char c)
+	 * bits [2..1] (CHRL):
+	 *  0xb : 8 data bits
+	 *  10b : 7 data bits
+	 *  11b : 6 data bits
+	 */
+
+	switch (databits) {
+	default:
+	case UART_CFG_DATA_BITS_8:
+		mode_reg |= XUARTPS_MR_CHARLEN_8_BIT;
+		break;
+	case UART_CFG_DATA_BITS_7:
+		mode_reg |= XUARTPS_MR_CHARLEN_7_BIT;
+		break;
+	case UART_CFG_DATA_BITS_6:
+		mode_reg |= XUARTPS_MR_CHARLEN_6_BIT;
+		break;
+	}
+
+	return mode_reg;
+}
+
+/**
+ * @brief Converts a flow control enum value to a Modem Control
+ *        Register bit mask.
+ *
+ * Converts a value of an enumeration type provided by the driver
+ * framework for the configuration of the UART's flow control
+ * setting into a bit mask within the Modem Control Register.
+ *
+ * It is assumed that the Modem Control Register contents that are
+ * being modified within this function come with the bits modified
+ * by this function already masked out by the caller.
+ *
+ * @param modemcr_reg Modem Control Register contents to which
+ *                    the flow control configuration shall be added.
+ * @param hwctrl Enumeration value to be converted to a bit mask.
+ *
+ * @return Modem Control Register contents to which the data bit
+ *         setting has been added.
+ */
+static inline u32_t uart_xlnx_ps_cfg2ll_hwctrl(
+	u32_t modemcr_reg,
+	enum uart_config_flow_control hwctrl)
+{
+	/*
+	 * Translate the new flow control configuration to the modem
+	 * control register's bit [5] (FCM):
+	 *  0b : no flow control
+	 *  1b : RTS/CTS
+	 */
+
+	if (hwctrl == UART_CFG_FLOW_CTRL_RTS_CTS) {
+		modemcr_reg |= XUARTPS_MODEMCR_FCM_RTS_CTS;
+	} else { /* effectively: UART_CFG_FLOW_CTRL_NONE */
+		modemcr_reg |= XUARTPS_MODEMCR_FCM_NONE;
+	}
+
+	return modemcr_reg;
+}
+
+/**
+ * @brief Configures the UART device at run-time.
+ *
+ * Configures the UART device at run-time according to the
+ * configuration data provided by the caller.
+ *
+ * @param dev UART device struct
+ * @param cfg The configuration parameters to be applied.
+ *
+ * @return 0 if the configuration completed successfully, ENOTSUP
+ *         error if an unsupported configuration parameter is detected.
+ */
+static int uart_xlnx_ps_configure(struct device *dev,
+	const struct uart_config *cfg)
+{
+	struct uart_xlnx_ps_dev_config *dev_cfg =
+	(struct uart_xlnx_ps_dev_config *)DEV_CFG(dev);
+
+	u32_t reg_base    = dev_cfg->uconf.regs;
+	u32_t mode_reg    = 0;
+	u32_t modemcr_reg = 0;
+
+	/* Controller supports only RTS/CTS flow control */
+	if (cfg->flow_ctrl != UART_CFG_FLOW_CTRL_NONE) {
+		if (cfg->flow_ctrl != UART_CFG_FLOW_CTRL_RTS_CTS) {
+			return -ENOTSUP;
+		}
+	}
+
+	/* Controller doesn't support 0.5 stop bits */
+	if (cfg->stop_bits == UART_CFG_STOP_BITS_0_5) {
+		return -ENOTSUP;
+	}
+
+	/* Controller doesn't support 5 or 9 data bits */
+	if (cfg->data_bits == UART_CFG_DATA_BITS_5
+		|| cfg->data_bits == UART_CFG_DATA_BITS_9) {
+		return -ENOTSUP;
+	}
+
+	/* Disable the controller before modifying any config registers */
+
```

Comment:
Nit: These empty lines after one line comments should not be necessary (I see that you are not adding an empty line after one line comments above -- so we should stick with either one of them to be consistent).